cmd/goal/application.go

```diff
@@ -120,6 +120,7 @@ func init() {
 
 	methodAppCmd.Flags().StringVar(&method, "method", "", "Method to be called")
 	methodAppCmd.Flags().StringArrayVar(&methodArgs, "arg", nil, "Args to pass in for calling a method")
+	methodAppCmd.Flags().StringVar(&onCompletion, "on-completion", "NoOp", "OnCompletion action for application transaction")
```

Comment:
This is the important part that was missing. Everything else is just a name change from `createOnCompletion` to `onCompletion`

Reply:
Yes, I think this is what cobra should do to have a default `NoOp` for onCompletion in method.